### PR TITLE
Branch v1.4 implementation for active person

### PR DESCRIPTION
**Pull request by Droffilc13**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -271,6 +272,46 @@ A `UniqueAssignmentList` stores a list of `Assignment` and prevents duplicates.
     * Pros: Divides cleanly all the different types of status and intended behaviour and make it very easy to add new status with few adjustments by creating another subclass.
     * Cons: The code length is very long due to all the subclasses of status and may not be optimal for Status class with very few status types.
 
+### Keeping track on person whose assignments are displayed feature
+
+#### Implementation
+
+The `activePerson` field is a reference to a `Person`object within `UniquePersonList`, wrapped in an `Optional`, whose `Assignment` objects are stored in `UniqueAssignmentList` within `AddressBook`. It forms the bedrock of many commands that involves `Assignment` in `UniqueAssignmentList`.
```
Comment (qinguorui2001) [inline docs/DeveloperGuide.md:279]: `Person` [SPACE]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -271,6 +272,46 @@ A `UniqueAssignmentList` stores a list of `Assignment` and prevents duplicates.
...
+
+`activePerson` field is wrapped in an `Optional` because there is a possibility that there are no `activePerson` which means that there are no `Person` objects whose assignment list is currently stored in the `UniqueAssignmentList`.
```
Comment (qinguorui2001) [inline docs/DeveloperGuide.md:281]: there IS no

Comment (qinguorui2001) [reply docs/DeveloperGuide.md:281]: there is no person object

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -271,6 +272,46 @@ A `UniqueAssignmentList` stores a list of `Assignment` and prevents duplicates.
...
+
+**Aspect: How undo & redo executes:**
+
+* **Alternative 1 (current choice): use a reference to point to `person` whose assignments should be displayed** 
+  * Pros: Easy to implement.
+  * Cons: Ignoring the property of `UniqueAssignmentList` that prevents duplicate `Assignment` from being stored, this method will be limited to displaying a particular person's assignments. Difficult to extend to displaying `Assignment` objects of multiple `Person` objects.
+
+* **Alternative 2:** `Person` class store an additional attribute `boolean isActivePerson`.
+  * Pros: Can toggle between multiple persons.
+  * Cons: `isActivePerson` may not be a suitable property of `Person` class since it may not be the responsibility of `Person` to remember whether it is the `activePerson`.
```
Comment (qinguorui2001) [inline docs/DeveloperGuide.md:314]: if change field name to `isActive` then it's possible to be a property of a person, so i think the con is not exact,

**Review by qinguorui2001 — CHANGES_REQUESTED** [review]
just change a few

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Droffilc13) [reply docs/DeveloperGuide.md:314]: hmm, but I think the responsibility shouldn't be on the person to remmeber if he/she is the active person. rather the list that contains it?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Droffilc13) [reply docs/DeveloperGuide.md:279]: good catch

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Droffilc13) [reply docs/DeveloperGuide.md:281]: changed!

**Pull request closed without merging**